### PR TITLE
Get versioninfo from the git repository

### PR DESCRIPTION
**Pull request by mrvdb**
Sets value of softwareVer during build

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,10 @@
+GITREV=`git describe --tags`
```
Comment (thebaer) [inline Makefile]: Maybe something like this instead, so we can keep `softwareVer` without the **v** included? (It's used in other parts of the app where we don't want the **v**, including in filenames of the archives we release.)

```suggestion
GITREV=`git describe --tags | cut -c 2-`
```

File: app.go [hunk 2]
```diff
@@ -33,10 +33,12 @@ const (
 
 	serverSoftware = "WriteFreely"
 	softwareURL    = "https://writefreely.org"
-
-	softwareVer = "0.3"
 )
 
+// Software version can be set from git env using -ldflags
+var	softwareVer = "v0.3"
```
Comment (thebaer) [inline app.go]: Could you format this file with `gofmt`? And if we remove the **v** from `$(GITREV)` as mentioned before, we can remove it here.

File: app.go [hunk 3]
```diff
@@ -133,7 +135,7 @@ func pageForReq(app *app, r *http.Request) page.StaticPage {
 	p := page.StaticPage{
 		AppCfg:  app.cfg.App,
 		Path:    r.URL.Path,
-		Version: "v" + softwareVer,
```
Comment (thebaer) [inline app.go]: We can leave this as-is if we remove the **v** from `$(GITREV)` as mentioned before.

**Comment by thebaer** [comment]
Looks great, thanks!

**Comment by mrvdb** [comment]
whoohoo!

**Pull request merged**